### PR TITLE
[GUI] Less GUI locks, IBD state cached.

### PR DESCRIPTION
**Pull request by furszy**
We were locking `cs_main`, calling `IsInitialBlockDownload` on every new row inserted in the transaction table model.

This PR caches the IBD state and updates it on every new block from the back-end without lock `cs_main` at all.

### CODE REVIEW

File: src/qt/clientmodel.h [hunk 1]
```diff
@@ -80,9 +83,10 @@ class ClientModel : public QObject
     QString formatClientStartupTime() const;
     QString dataDir() const;
 
-    void setCacheTip(const CBlockIndex* const tip) { cacheTip = tip; };
-    void setCacheReindexing(bool reindex) { cachedReindexing = reindex; };
-    void setCacheImporting(bool import) { cachedImporting = import; };
+    void setCacheTip(const CBlockIndex* const tip) { cacheTip = tip; }
+    void setCacheReindexing(bool reindex) { cachedReindexing = reindex; }
+    void setCacheImporting(bool import) { cachedImporting = import; }
+    void setCacheInitialSync(const bool& _initialSync) { cachedInitialSync = _initialSync; }
```
[the remark this answers is not in the document]
Comment (furszy) [reply src/qt/clientmodel.h]: updated.

File: src/qt/clientmodel.h [hunk 2]
```diff
@@ -69,6 +69,9 @@ class ClientModel : public QObject
     //! Return true if core is doing initial block download
     bool inInitialBlockDownload() const;
     //! Return true if core is importing blocks
+    bool IsImporting() const { return cachedImporting; }
+    //! Return true if it's reindexing.
+    bool IsReindexing() const { return cachedReindexing; }
```
Comment (random-zebra) [inline src/qt/clientmodel.h]: It seems we are not using these functions anywhere.

Comment (furszy) [reply src/qt/clientmodel.h]: Not here, have another branch using them. Will move the commit there.

Comment (furszy) [reply src/qt/clientmodel.h]: done.

**Pull request merged**